exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logicmonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/logicmonitorexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/logicmonitor/lm-data-sdk-go/api/logs"
+	"github.com/logicmonitor/lm-data-sdk-go/utils"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logExporter struct {
+	config          *Config
+	logger          *zap.Logger
+	logIngestClient *logs.LMLogIngest
+	settings        component.TelemetrySettings
+}
+
+const (
+	hostname         = "hostname"
+	hostnameProperty = "system.hostname"
+)
+
+// Create new logicmonitor logs exporter
+func newLogsExporter(cfg component.Config, set component.ExporterCreateSettings) *logExporter {
+	oCfg := cfg.(*Config)
+
+	return &logExporter{
+		config:   oCfg,
+		logger:   set.Logger,
+		settings: set.TelemetrySettings,
+	}
+}
+
+func (e *logExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
```

Comment:
I think batching should be enabled by default?

Reply:
According to contributing guidelines `Avoid introducing batching, retries or worker pools directly on receivers and exporters. Typically, these are general cases that can be better handled via processors (that also can be reused by other receivers and exporters).`  and suggestion by @bogdandrutu , we have disabled batching in logicmonitorexporter.

Reply:
Ah. Weird. I would have assumed that sending all the contents of your logs at once was the right idea here.